tests/test_qubit_device.py

```diff
@@ -326,30 +326,36 @@ def test_parameters_accessed_outside_execution_context(self, mock_qubit_device):
 class TestExtractStatistics:
     """Test the statistics method"""
 
-    def test_observables_deprecated(self, mock_qubit_device_extract_stats, monkeypatch):
+    def test_list_of_observables_deprecated(self, mock_qubit_device_extract_stats, monkeypatch):
         """Test that using a list of observables as an argument is deprecated."""
-        with monkeypatch.context() as m:
-            dev = mock_qubit_device_extract_stats()
-            with pytest.warns(
-                UserWarning,
-                match="Using a list of observables in ``QubitDevice.statistics`` is",
-            ):
-                dev.statistics([])
-            with pytest.warns(
-                UserWarning,
-                match="Using a list of observables in ``QubitDevice.statistics`` is",
-            ):
-                dev.statistics(observables=[])
-            qscript = QuantumScript()
-            with pytest.warns(
-                UserWarning,
-                match="Using a list of observables in ``QubitDevice.statistics`` is",
-            ):
-                dev.statistics([], circuit=qscript)
-            with pytest.raises(
-                ValueError, match="Please provide a circuit into the statistics method"
-            ):
-                dev.statistics()
+        dev = mock_qubit_device_extract_stats()
+        with pytest.warns(
+            UserWarning,
+            match="Using a list of observables in ``QubitDevice.statistics`` is",
+        ):
+            dev.statistics([])
+
+    def test_observables_kwarg_deprecated(self, mock_qubit_device_extract_stats, monkeypatch):
+        """Test that using observables instead of circuit is deprecated."""
+        dev = mock_qubit_device_extract_stats()
+        with pytest.warns(
+            UserWarning,
+            match="The ``observables`` argument in ``QubitDevice.statistics`` is deprecated. ",
+        ):
+            dev.statistics(observables=[])
+        qscript = QuantumScript()
+        with pytest.warns(
+            UserWarning,
+            match="The ``observables`` argument in ``QubitDevice.statistics`` is deprecated. ",
+        ):
+            dev.statistics([], observables=qscript)
```

Comment:
Can you test:
```python
dev.statistics([], circuit=qscript)
```
and 
```python
dev.statistics(qscript, observables=[])
```
too?

Reply:
Just realised that using `dev.statistics([], circuit=qscript)` might fail.

We must ensure that the old signature still works:
```python
def statistics(observables, shot_range=None, bin_size=None, circuit=None)
```

Reply:
What would you want it to do in these cases? 

Currently the first one raises an error (`TypeError: QubitDevice.statistics() got multiple values for argument 'circuit'`), and the second warns that `The ``observables`` argument in ``QubitDevice.statistics`` is deprecated. Please use ``circuit`` instead.`

Reply:
That is why the order of the arguments was mantained (sorry I just remembered). We will need to move `observables` back to the first argument.